### PR TITLE
Fix codespell to work for Ubuntu 20 and 18

### PR DESCRIPTION
**Pull request by akhilveeraghanta**
### Description
Codespell's dictionaries are different between the versions on ubuntu 20 and 18, we freeze the dictionary here to the most up-to-date one that both versions support.

### Testing Done
Manual testing

### Resolved Issues
N/A

### Length Justification
N/A

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: formatting_scripts/fix_formatting.sh [hunk 1]
```diff
@@ -71,12 +71,14 @@ function run_black_formatting () {
 }
 
 function run_code_spell(){
+    curl https://raw.githubusercontent.com/codespell-project/codespell/v1.14.0/codespell_lib/data/dictionary.txt --output /tmp/dictionary.txt
```
Comment (jonathanlew) [inline formatting_scripts/fix_formatting.sh]: what happens if curl fails?

Comment (akhilveeraghanta) [reply formatting_scripts/fix_formatting.sh]: hello

Comment (akhilveeraghanta) [reply formatting_scripts/fix_formatting.sh]: you would probably just run the script again?

Comment (jonathanlew) [reply formatting_scripts/fix_formatting.sh]: This is the error if curl fails, which is confusing. Can we check that curl worked before calling codespell?
```
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100    14  100    14    0     0    168      0 --:--:-- --:--:-- --:--:--   168
Fixing spelling...

Traceback (most recent call last):
  File "/usr/bin/codespell", line 558, in <module>
    sys.exit(main(*sys.argv))
  File "/usr/bin/codespell", line 505, in main
    build_dict(options.dictionary)
  File "/usr/bin/codespell", line 284, in build_dict
    [key, data] = line.split('->')
ValueError: not enough values to unpack (expected 2, got 1)
Traceback (most recent call last):
  File "/usr/bin/codespell", line 558, in <module>
    sys.exit(main(*sys.argv))
  File "/usr/bin/codespell", line 505, in main
    build_dict(options.dictionary)
  File "/usr/bin/codespell", line 284, in build_dict
    [key, data] = line.split('->')
ValueError: not enough values to unpack (expected 2, got 1)
Traceback (most recent call last):
  File "/usr/bin/codespell", line 558, in <module>
    sys.exit(main(*sys.argv))
  File "/usr/bin/codespell", line 505, in main
    build_dict(options.dictionary)
  File "/usr/bin/codespell", line 284, in build_dict
    [key, data] = line.split('->')
ValueError: not enough values to unpack (expected 2, got 1)
Traceback (most recent call last):
  File "/usr/bin/codespell", line 558, in <module>
    sys.exit(main(*sys.argv))
  File "/usr/bin/codespell", line 505, in main
    build_dict(options.dictionary)
  File "/usr/bin/codespell", line 284, in build_dict
    [key, data] = line.split('->')
ValueError: not enough values to unpack (expected 2, got 1)
Traceback (most recent call last):
  File "/usr/bin/codespell", line 558, in <module>
    sys.exit(main(*sys.argv))
  File "/usr/bin/codespell", line 505, in main
    build_dict(options.dictionary)
  File "/usr/bin/codespell", line 284, in build_dict
    [key, data] = line.split('->')
ValueError: not enough values to unpack (expected 2, got 1)
```

Comment (akhilveeraghanta) [reply formatting_scripts/fix_formatting.sh]: oof, fixed!

File: formatting_scripts/fix_formatting.sh [hunk 2]
```diff
@@ -71,12 +71,19 @@ function run_black_formatting () {
 }
 
 function run_code_spell(){
+    http_code=$(curl -sw '%{http_code}' https://raw.githubusercontent.com/codespell-project/codespell/v1.14.0/codespell_lib/data/dictionary.txt --output /tmp/dictionary.txt)
+
+    if [[ "$http_code" != 200 ]]; then
+        printf "\n***Failed to download codespell dictionay!***\n\n"
```
Comment (jonathanlew) [inline formatting_scripts/fix_formatting.sh]:
```suggestion
        printf "\n***Failed to download codespell dictionary!***\n\n"
```
ironic

**Review by jonathanlew — APPROVED** [review]

**Review by itsarune — APPROVED** [review]
👍

**Pull request merged**